api/codes.go

```diff
@@ -35,10 +55,17 @@ func (c *CodeListAPI) getCodes(w http.ResponseWriter, r *http.Request) {
 		codes.Items[i] = item
 	}
 
-	count := len(codes.Items)
+	sort.Slice(codes.Items, func(i, j int) bool {
+		return codes.Items[i].ID < codes.Items[j].ID
+	})
+
+	slicedResults := codesSlice(codes.Items, offset, limit)
```

Comment:
There is code above on line 49 that iterates the codes and updates the links. I think this should be done after the codes have been sorted and sliced, as there is no need to update links on codes that are not returned. This should be done for all similar endpoints that update the results.